gate-x509/src/main/groovy/com/netflix/spinnaker/gate/security/x509/X509AuthenticationUserDetailsService.groovy

```diff
@@ -243,4 +238,34 @@ class X509AuthenticationUserDetailsService implements AuthenticationUserDetailsS
       it.find { it.toString() == RFC822_NAME_ID }
     }?.get(1)
   }
+
+  /**
+   * Extract identity from an x509 certificate.
+   *
+   * Strategies (in priority order):
+   * - X509UserIdentifierExtractor (when supplied)
+   * - The certificates RFC822 name
+   * - The certificates common name
+   *
+   * @param x509Certificate
+   * @return Extracted identity or null if none available
+   */
+  String identityFromCertificate(X509Certificate x509Certificate) {
```

Comment:
I feel like I'd written something similar elsewhere in Gate? Is this happening earlier than that?

Reply:
Most of this is really just teased out from above and made callable from outside of the spring security filter chain.

Considering the x509 common name is new behavior but maps to what `token.principal?.toString()` would have provided.